### DISCUSSION TITLE
Add support for time-based exercises

### DISCUSSION THREAD

**Discussion by astashov** [astashov/liftosaur#132] [Ideas]
There's currently a workoaround for it, to use reps as seconds, so people do it like `Plank / 1x60` to do a 60s plank, but it's a bit awkward.

Would be nice to have proper support of seconds, and maybe built-in timer, that would record the duration of the exercise.

**Comment by yrinkel** [comment]
Yes, important point. 
not only for planks but for *timed set* exercises, where you do AMRAP but in a limited time window. 
Finally that would allow you to track Time Under Tension (TUT/TUL) for each exercise.

**Comment by galantra** [comment]
Suitable for logging cardio, too.

**Reply by bastislack** [reply]
It would be great if when the set is done and the following rest timer is also finished, the next set immediately starts. This way we could program interval training :)

**Comment by 13hannes11** [comment]
Would also be nice to have support for distance/time.

Use cases are rowing, running, etc.

Run 5 mins and track distance.
Run 1 km and track time.

**Comment by jm96441n** [comment]
Another use case is for things like farmers carry/yoke runs where's there's a time/weight/distance combination and for strongman events like log press amrap within a time frame.

**Comment by Finn-Gaughan** [comment]
Another use case would be for adding static stretches that require you to hold a certain position for a specific time. Like Toe Touches for 60 seconds x 3 sets.

Adding stretches onto the end of my workout would be so helpful without having to leave the app.

**Comment by MarcioFCMartins** [comment]
For me, transitioning to liftosaur isn't really an option right now since my other app has 11 years of tracked exercises, which occasionally include assault bike, farmers carries and other types of exercises which combine weight, time and distance. 

I currently use Fitnotes and quite like their implementation: when creating an exercise you can choose how it will be tracked from a combination of weights, reps, time and distance. I think this would be a great approach.

**Reply by AustinGrey** [reply]
Same here, fitnotes has my data, and I can't migrate yet because I'd lose all that fidelity

**Comment by xavierclarkvt** [comment]
Any update on this? I'm doing two sets of AMRAP plank and it'd be great to be able to log how long I've kept it up for. Adding an in-app stopwatch would be a great addition for this. Thanks!

**Comment by PokeYouInDaEye** [comment]
Not to scope-creep, but thinking through my use-cases over the years:
- Sandbag (weight) for reps
- Sandbag or sled for distance
- Sandbag or sled for duration
- Jump rope for duration
- Jump rope for reps/jumps
- Spints (distance or time) for reps
- Jumps (height or distance) for reps
- Boxing (eg, heavy bag) rounds (RPE?) for time

Secondly/related, the concept of time/rest is, IMHO, under-utilized in fitness apps and training.. eg, some programs have EMOM (every minute on the minute). I wonder if, both for this feature requests main function, and possibly a secondary value-add, having a third “column” next to rep/weight (or whatever metric[s] get added) could be (actual) rest between sets. For my experience (10+ years lifting), 4 sets of 10 reps with 30 seconds rest vs. 2.5 minutes rest is drastically different, enough to merit tracking as a key metric almost as much as sets/reps/weight (or distance/etc.).

**Comment by itpeters** [comment]
I would love to be able to use Liftosaur for things like 30-10-30 sets (30 second controlled negative, 10 controlled positive/negative reps, one more 30 second controlled negative). I currently use my interval timer app for these sets, but it would be great to have them in my Liftosaur program.

**Comment by YamadaDesigns** [comment]
For treadmill and elliptical, it would be awesome to be able to track not only time, but things like level/speed, incline, and distance.

**Comment by astashov** [comment]
This is done!

**Reply by fun4stuff** [reply]
How do you convert an exercise to times one? I do not see how. Thanks

**Reply by astashov** [reply]
Hey! You do that in the program. Like `Plank / 3x1 60s|30s`. Check the docs: https://www.liftosaur.com/doc/liftoscript, section "Set timers (active timers) and circuits"
